Sofa/Component/Topology/Mapping/src/sofa/component/topology/mapping/SubsetTopologicalMapping.cpp

```diff
@@ -941,6 +941,106 @@ void SubsetTopologicalMapping::updateTopologicalMappingTopDown()
             break;
         }
 
+        case core::topology::TETRAHEDRAADDED:
+        {
+        }
+
+        case core::topology::TETRAHEDRAREMOVED:
+        {
+            if (!d_handleTetrahedra.getValue()) break;
+            if (!toTetrahedronMod) toModel->getContext()->get(toTetrahedronMod);
+            if (!toTetrahedronMod) break;
+
+            // teRem is a pointer to the tetra elements that should be removed
+            // this change comes from the topology container in the source topology.
+            // So here, we have to update our data that maps between source and destination
+            // teD2S and teS2D, and also update the topology container of this model (the destination)
+            const TetrahedraRemoved *teRem = static_cast< const TetrahedraRemoved * >( topoChange );
+
+            // each tetra is an integer that specifies it's index in the source topology
+            sofa::type::vector<Index> tetra_array_buffer_source = teRem->getArray();
+
+            // we want to remove the elements in both source and destination
+            sofa::type::vector<Index> tetra_array_buffer_destination;
+            tetra_array_buffer_destination.reserve(tetra_array_buffer_source.size());
+
+            // for each tetra that should be removed, look up the correct tetra in the destination topology
+            for (auto &tetra_source : tetra_array_buffer_source)
```

Comment:
I would suggest to use tetra_source_id to show that we are working on a set of indices and not on tetrahedra arrays